template/generate_template.R

```diff
@@ -1,60 +1,109 @@
 
-# Note that weekly incidence forecasts are always from Sunday (included) to Saturday (included; this is target_end_date)
 
 # define the date on which forecasts are generated:
-forecast_date <- as.Date("2020-07-13")
+forecast_date <- as.Date("2021-02-01")
 
-locations <- read.csv("state_codes_germany.csv")
+locations <- read.csv("template/locations_eu.csv")
 
 dat <- data.frame(
+  scenario_id = "EXAMPLE",
   forecast_date = forecast_date,
-  target = NA,
-  target_end_date = NA,
-  location = "GM",
-  location_name	= "Germany",
-  type = NA,
+  target = "4 wk ahead inc case",
+  target_end_date = forecast_date + 5, # forecast_date = Mon, target = Sat
+  location = "EXAMPLE",
+  type = "point",
   quantile = NA,
-  value = NA
+  value = 1
 )
 
-tgs <- c(paste(-1:30, "day ahead inc death"),
-         paste(-1:30, "day ahead cum death"),
-         paste(-1:4, "wk ahead inc death"),
-         paste(-1:4, "wk ahead inc death"))
+tgs <- c(paste(1:4, "wk ahead inc case"),
+         paste(1:4, "wk ahead inc death"))
 
-end_dates <- c(forecast_date + (-1:30),
-               forecast_date + (-1:30),
-               forecast_date + c(-9, -2, 5, 12, 19, 26),
-               forecast_date + c(-9, -2, 5, 12, 19, 26))
+end_dates <- c(forecast_date + c(6, 13, 20, 27),
+               forecast_date + c(6, 13, 20, 27))
 
 quantiles <- c(0.01, 0.025, 1:19/20, 0.975, 0.99)
 
 weekdays(end_dates)
 
 for(loc in 1:nrow(locations)){
   for(t in seq_along(tgs)){
-    new_dat <- data.frame(forecast_date = forecast_date,
+    new_dat <- data.frame(scenario_id = "forecast",
+                          forecast_date = forecast_date,
                           target = tgs[t],
                           target_end_date = end_dates[t],
-                          location = locations$state_code[loc],
-                          location_name = locations$state_name[loc],
+                          location = locations$iso3c[loc],
                           type = c("point", rep("quantile", length(quantiles))),
                           quantile = c(NA, quantiles),
-                          value = NA)
+                          value = 1)
     dat <- rbind(dat, new_dat)
   }
 }
 
 
 dat$target_end_date <- as.Date(dat$target_end_date, origin = "1970-01-01")
-# set type = "observed" for past dates:
-dat$type[dat$type == "point" & dat$target_end_date <= forecast_date] <- "observed"
-dat <- subset(dat, type == "observed" | target_end_date > forecast_date)
+dat$forecast_date <- as.Date(dat$forecast_date, origin = "1970-01-01")
+
+write.csv(dat, file = paste0("data-processed/Template-ExampleModel/", forecast_date, "-Template-ExampleModel.csv"), row.names = FALSE)
+write.csv(dat, file = paste0("template/", forecast_date, "-Template-ExampleModel.csv"), row.names = FALSE)
+
+
+# Submission dates & epiweeks ------------------------------------
+
+forecast_date <- tibble::tibble(
+  forecast_date	= seq.Date(as.Date("2021-02-01"), as.Date("2022-01-02"), by = 7),
```

Comment:
I'm really uncomfortable with having a fixed end date here as this is bound to be forgotten when it comes to it but I realise this is hardcoded in a few places and will be a pain to change. At the least can we add a year?